ui/src/app/zones/views/ZoneDetails/ZoneDetailsHeader/ZonesDetailsHeader.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { mount } from "enzyme";
+import { Provider } from "react-redux";
+import { MemoryRouter, Route } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import ZoneDetailsHeader from "./ZoneDetailsHeader";
+
+import type { RootState } from "app/store/root/types";
+import {
+  zone as zoneFactory,
+  zoneState as zoneStateFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("ZoneDetailsHeader", () => {
+  let initialState: RootState;
+
+  beforeEach(() => {
+    initialState = rootStateFactory({
+      zone: zoneStateFactory({
+        errors: {},
+        loading: false,
+        loaded: true,
+        items: [
+          zoneFactory({
+            id: 1,
+            name: "zone-name",
+          }),
+        ],
+      }),
+    });
+  });
+
+  it("displays zone name in header if one exists", () => {
+    const state = { ...initialState };
```

Comment:
I'm not sure this is necessary as you're getting a new `initialState` on every test run and a spread doesn't actually 'clone' any nested objects.

Reply:
That's probably our fault as that's done in some old tests from before we hand the factories.